File: StartHere/src/main/java/com/lambdaschool/starthere/services/BookService.java
package com.lambdaschool.starthere.services;

import com.lambdaschool.starthere.models.Book;
import org.springframework.data.domain.Pageable;

import java.util.ArrayList;

public interface BookService {

    ArrayList<Book> findAll(Pageable pageable);
    void update(long id, Book book);
    void delete(long id);
    void setBookAuthor(long bookid, long authorid);

}

File: sprint14/src/main/kotlin/com/lambdaschool/starthere/services/BookService.kt
package com.lambdaschool.starthere.services

import com.lambdaschool.starthere.models.Book
import org.springframework.data.domain.Pageable

import java.util.ArrayList

interface BookService {

    fun findAll(pageable: Pageable): ArrayList<Book>
    fun update(id: Long, book: Book)
    fun delete(id: Long)
    fun setBookAuthor(bookid: Long, authorid: Long)

}
